src/app/modules/vault/modules/organization-vault/organization-vault.component.ts

```diff
@@ -57,6 +58,11 @@ export class OrganizationVaultComponent implements OnInit, OnDestroy {
   trashCleanupWarning: string = null;
   activeFilter: VaultFilter = new VaultFilter();
 
+  // This is a hack to avoid redundant api calls that fetch OrganizationVaultFilterComponent collections
+  // When it makes sense to do so we should leverage some other communication method for change events that isn't directly tied to the query param for organizationId
+  // i.e. exposing the VaultFiltersService to the OrganizationSwitcherComponent to make relevant updates from a change event instead of just depending on the router
+  skipFilterCollectionRefresh = true;
```

Comment:
I suggest we rename this `firstLoaded`, which is the naming we use elsewhere for this pattern. It communicates better how the flag works.

I think this is an acceptable short-term fix, but I think it does mean we should try to implement your refactor soon.

Reply:
Also worth noting this issue is also present for the individual vault - it hits local data, so it's less important, but ideally any long-term fix here should apply to both vaults.